### ISSUE TITLE
Validator always fails objects with codes

### ISSUE THREAD

**Issue by nrpeterson** [jamesagnew/hapi-fhir#136]
The DSTU2 structures seem to always fail to validate codes on Observations.  For instance, I copied the following example JSON file from FHIR's website at http://hl7-fhir.github.io/observation-example.json.html:

```json
{
  "resourceType": "Observation",
  "id": "example",
  "text": {
    "status": "generated",
    "div": "<div>!-- Snipped for Brevity --></div>"
  },
  "code": {
    "coding": [
      {
        "system": "http://loinc.org",
        "code": "3141-9",
        "display": "Weight Measured"
      },
      {
        "system": "http://snomed.info/sct",
        "code": "27113001",
        "display": "Body weight"
      },
      {
        "system": "http://acme.org/devices/clinical-codes",
        "code": "body-weight",
        "display": "Body Weight"
      }
    ]
  },
  "valueQuantity": {
    "value": 185,
    "units": "lbs",
    "system": "http://unitsofmeasure.org",
    "code": "[lb_av]"
  },
  "status": "final",
  "reliability": "ok",
  "subject": {
    "reference": "Patient/example"
  }
}
```

With this defined as `resourceStr` in Scala, I can do
```scala
val resource = jsonParser.parseResource(resourceStr)
val result = validator.validateWithResult(resource)
println(jsonParser.encodeResourceToString(result.getOperationOutcome())
```
and the result is
```json
{
    "resourceType":"OperationOutcome",
    "issue":[
        {
            "severity":"error",
            "details":"cvc-complex-type.2.4.a: Invalid content was found starting with element 'code'. One of '{\"http://hl7.org/fhir\":contained, \"http://hl7.org/fhir\":extension, \"http://hl7.org/fhir\":modifierExtension, \"http://hl7.org/fhir\":name}' is expected.",
            "location":[
                "Line[1] Col[199]"
            ]
        }
    ]
}
```

**Comment by Tia** [comment]
If you're on version 0.9 , it's still looking for "name" instead of "code" .  I believe this is fixed in 1.0-snapshot.

Tia

**Comment by nrpeterson** [comment]
Yep, upgrading absolutely fixed the issue.  Thanks, @Tia !
